packages/config/src/ast-utils/addToCypressConfig.ts

```diff
@@ -0,0 +1,171 @@
+  filePath: string
+  projectId: string
+}
+
+export async function addProjectIdToCypressConfig (options: AddProjectIdToCypressConfigOptions) {
+  try {
+    let result = await fs.readFile(options.filePath, 'utf8')
+    const toPrint = await addToCypressConfig(options.filePath, result, t.objectProperty(
+      t.identifier('projectId'),
+      t.identifier(options.projectId),
+    ))
+
+    await fs.writeFile(options.filePath, maybeFormatWithPrettier(toPrint, options.filePath))
+
+    return {
+      result: 'MERGED',
+    }
+  } catch (e) {
+    return {
+      result: 'NEEDS_MERGE',
+      error: e,
+    }
+  }
+}
+
+export interface AddToCypressConfigResult {
+  result: 'ADDED' | 'MERGED' | 'NEEDS_MERGE'
+  error?: Error
+  codeToMerge?: string
+}
+
+export interface AddTestingTypeToCypressConfigOptions {
+  filePath: string
+  info: ASTComponentDefinitionConfig | {
+    testingType: 'e2e'
+  }
+}
+
+export async function addTestingTypeToCypressConfig (options: AddTestingTypeToCypressConfigOptions): Promise<AddToCypressConfigResult> {
+  const toAdd = options.info.testingType === 'e2e' ? addE2EDefinition() : addComponentDefinition(options.info)
+
+  try {
+    let result: string | undefined = undefined
+    let resultStatus: 'ADDED' | 'MERGED' = 'MERGED'
+
+    try {
+      result = await fs.readFile(options.filePath, 'utf8')
+    } catch (e) {
+      // If we can't find the file, or it's an empty file, let's create a new one
+    }
+
+    const pathExt = path.extname(options.filePath)
+
+    // If for some reason they have deleted the contents of the file, we want to recover
+    // gracefully by adding some default code to use as the AST here, based on the extension
+    if (!result || result.trim() === '') {
+      resultStatus = 'ADDED'
+      result = getEmptyCodeBlock(pathExt as OutputExtension)
+    }
+
+    const toPrint = await addToCypressConfig(options.filePath, result, toAdd)
+
+    await fs.writeFile(options.filePath, maybeFormatWithPrettier(toPrint, options.filePath))
+
+    return {
+      result: resultStatus,
+    }
+  } catch (e) {
+    return {
+      result: 'NEEDS_MERGE',
+      error: e,
+      codeToMerge: print(toAdd).code,
+    }
+  }
+}
+
+type OutputExtension = '.ts' | '.mjs' | '.js'
+
+// Necessary to handle the edge case of them deleting the contents of their Cypress
+// config file, just before we merge in the testing type
+function getEmptyCodeBlock (outputType: OutputExtension) {
+  if (outputType === '.ts' || outputType === '.mjs') {
+    return dedent`
+      import { defineConfig } from 'cypress'
+
+      export default defineConfig({
+        
+      })
+    `
+  }
+
+  return dedent`
+    const { defineConfig } = require('cypress')
+
+    module.exports = defineConfig({
+      
+    })
+  `
+}
+
+function maybeFormatWithPrettier (code: string, filePath: string) {
+  try {
+    const prettierImportPath = require.resolve('prettier', { paths: [path.dirname(filePath)] })
```

Comment:
Curious why we do the require syntax here - we ship our own version of prettier in the bundle from the looks of things, should we just use that? My understanding is we'd like to use their local version, if they've got one, but if they don't, should we fallback to the one we are shipping w/ the binary?

Reply:
> we ship our own version of prettier in the bundle from the looks of things, should we just use that?

We’re trying to avoid that moving forward, it’s like 30mb